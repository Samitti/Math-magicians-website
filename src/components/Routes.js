import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import Home from './Home';
import Quote from './Quote';
import Navbar from './Navbar';
import App from './App';

const Routes = () => (
  <BrowserRouter>
    <Navbar />
    <Switch>
      <Route exact path="/" component={Home} />
      <Route exact path="/App" component={App} />
      <Route exact path="/Quote" component={Quote} />
    </Switch>
  </BrowserRouter>
);

export default Routes;
