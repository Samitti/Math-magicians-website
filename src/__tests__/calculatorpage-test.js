import React from 'react';
import ReactDOM from 'react-dom';
import Routes from '../components/Routes';

it('checks the snapshot of calculatorpage', () => {
    const div = document.createElement('div')
    document.body.appendChild(div)
    ReactDOM.render(<Routes />, div)
    const calculatorpage = document.body.querySelector('#calculatorpage')
    calculatorpage.click()
    expect(div).toMatchSnapshot();
});
