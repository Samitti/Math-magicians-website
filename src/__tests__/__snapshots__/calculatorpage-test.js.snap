// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks the snapshot of calculatorpage 1`] = `
<div>
  <div
    class="topAll"
  >
    <h1>
      Math Magicians
    </h1>
    <nav
      class="navLink"
    >
      <p>
        <a
          href="/"
          id="homepage"
          style="color: green;"
        >
          Home
        </a>
      </p>
      <p
        class="linkSplit"
      >
        |
      </p>
      <p>
        <a
          href="/App"
          id="calculatorpage"
          style="color: green;"
        >
          Calculator
        </a>
      </p>
      <p
        class="linkSplit"
      >
        |
      </p>
      <p>
        <a
          href="/Quote"
          id="quotepage"
          style="color: green;"
        >
          Quote
        </a>
      </p>
    </nav>
  </div>
  <div
    class="calcContainer"
  >
    <h2>
      Let’s do some math!
    </h2>
    <div
      class="App"
    >
      <div
        class="display"
      >
        0
      </div>
      <div
        class="button-panel"
      >
        <div>
          <button
            class="normal"
            style="background-color: gray;"
            type="button"
          >
            AC
          </button>
          <button
            class="normal"
            style="background-color: gray;"
            type="button"
          >
            +/-
          </button>
          <button
            class="normal"
            style="background-color: gray;"
            type="button"
          >
            %
          </button>
          <button
            class="normal"
            style="background-color: orange;"
            type="button"
          >
            ÷
          </button>
        </div>
        <div>
          <button
            class="normal"
            style="background-color: gray;"
            type="button"
          >
            7
          </button>
          <button
            class="normal"
            style="background-color: gray;"
            type="button"
          >
            8
          </button>
          <button
            class="normal"
            style="background-color: gray;"
            type="button"
          >
            9
          </button>
          <button
            class="normal"
            style="background-color: orange;"
            type="button"
          >
            x
          </button>
        </div>
        <div>
          <button
            class="normal"
            style="background-color: gray;"
            type="button"
          >
            4
          </button>
          <button
            class="normal"
            style="background-color: gray;"
            type="button"
          >
            5
          </button>
          <button
            class="normal"
            style="background-color: gray;"
            type="button"
          >
            6
          </button>
          <button
            class="normal"
            style="background-color: orange;"
            type="button"
          >
            -
          </button>
        </div>
        <div>
          <button
            class="normal"
            style="background-color: gray;"
            type="button"
          >
            1
          </button>
          <button
            class="normal"
            style="background-color: gray;"
            type="button"
          >
            2
          </button>
          <button
            class="normal"
            style="background-color: gray;"
            type="button"
          >
            3
          </button>
          <button
            class="normal"
            style="background-color: orange;"
            type="button"
          >
            +
          </button>
        </div>
        <div>
          <button
            class="wide"
            style="background-color: gray;"
            type="button"
          >
            0
          </button>
          <button
            class="normal"
            style="background-color: gray;"
            type="button"
          >
            .
          </button>
          <button
            class="normal"
            style="background-color: orange;"
            type="button"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
